### PR TITLE
Enable cargo-spellcheck for blockchain/chain

### LINKED ISSUE #1686: Enable `cargo-spellcheck` for `blockchain/chain`
**Issue summary**

Make sure that `cargo spellcheck` can run without errors in `blockchain/chain`. For details, see the tracking issue.

**Other information and links**

### PR DESCRIPTION
**Pull request by elmattic**
**Summary of changes**
Changes introduced in this pull request:
- Fix spellcheck errors

**Reference issue to close (if applicable)**

Closes #1686

**Other information and links**

### CODE REVIEW

File: blockchain/chain/src/store/base_fee.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use std::collections::HashSet;
 /// Used in calculating the base fee change.
 pub const BLOCK_GAS_TARGET: i64 = BLOCK_GAS_LIMIT / 2;
 
-/// Limits gas base fee change to 12.5% of the change.
+/// Limits gas base fee change to 12.5 percent of the change.
```
Comment (lemmih) [inline blockchain/chain/src/store/base_fee.rs:16]: hunspell doesn't understand `%`?

File: blockchain/chain/src/store/chain_store.rs [hunk 2]
```diff
@@ -109,7 +109,7 @@ where
         cs
     }
 
-    /// Sets heaviest tipset within ChainStore and store its tipset cids under HEAD_KEY
+    /// Sets heaviest tipset within `ChainStore` and store its tipset CIDs under `HEAD_KEY`
```
Comment (lemmih) [inline blockchain/chain/src/store/chain_store.rs:112]: Maybe we can re-word the docs to talk about keys rather than CIDs.
```suggestion
    /// Sets heaviest tipset within `ChainStore` and store its tipset keys under `HEAD_KEY`
```

File: .config/forest.dic [hunk 3]
```diff
@@ -19,3 +19,12 @@ RPC
 parameterize/D
 timestamp/SM
 synchronizer
+pointer/S
+validator/S
+threadsafe
+blockchain
+CID
+CIDs
```
Comment (lemmih) [inline .config/forest.dic:27]:
```suggestion
CID/SM
```

File: blockchain/chain/src/store/base_fee.rs (hunk 1 above)
Comment (elmattic) [reply blockchain/chain/src/store/base_fee.rs:16]: No and didn't find a way to make it work.

File: .config/forest.dic (hunk 3 above)
Comment (elmattic) [reply .config/forest.dic:27]: I tried this at first and it didn't work. Looks like the suffix rules don't work for uppercase words.

File: blockchain/chain/src/store/chain_store.rs (hunk 2 above)
Comment (elmattic) [reply blockchain/chain/src/store/chain_store.rs:112]: Good idea.

File: blockchain/chain/src/store/base_fee.rs (hunk 1 above)
Comment (elmattic) [reply blockchain/chain/src/store/base_fee.rs:16]: Maybe using similar `hunspell.quirks` settings? [like here](https://gitlab.parity.io/parity/mirrors/polkadot/-/blob/master/scripts/ci/gitlab/spellcheck.toml)

File: .config/forest.dic (hunk 3 above)
Comment (elmattic) [reply .config/forest.dic:27]: Weird, they managed to make it work [here](https://gitlab.parity.io/parity/mirrors/polkadot/-/blob/master/scripts/ci/gitlab/lingua.dic) (ie `VRF/SM`).

File: .config/forest.dic [hunk 4]
```diff
@@ -19,3 +19,12 @@ RPC
...
+tipsetkey/S
+Secp256k1
+IPLD
```
Comment (LesnyRumcajs) [inline .config/forest.dic:30]: can we have this dictionary sorted? can be in a different PR.

File: .config/forest.dic [hunk 5]
```diff
@@ -19,3 +19,12 @@ RPC
 parameterize/D
 timestamp/SM
 synchronizer
+pointer/S
+validator/S
+threadsafe
```
Comment (LesnyRumcajs) [inline .config/forest.dic:24]: I believe common writing is `thread-safe` so we shouldn't have an exception here

File: .config/spellcheck.toml [hunk 6]
```diff
@@ -10,7 +10,12 @@ tokenization_splitchars = "\",;:.!?#(){}[]|/_-‒'`&@§¶…"
 extra_dictionaries      = ["forest.dic"]
 
 [hunspell.quirks]
-transform_regex     = []
+transform_regex     = [
+  # 10.7%
+  "^[0-9_]+(?:\\.[0-9]*)?(x|%)$",
```
Comment (LesnyRumcajs) [inline .config/spellcheck.toml:15]: job security :laughing:

File: blockchain/chain/src/store/chain_store.rs [hunk 7]
```diff
@@ -664,16 +664,16 @@ where
     Ok(ts)
 }
 
-/// Helper to ensure consistent Cid -> db key translation.
+/// Helper to ensure consistent CID to db key translation.
```
Comment (LesnyRumcajs) [inline blockchain/chain/src/store/chain_store.rs:667]: probably `db` should be `DB`, no strong feelings though

**Review by LesnyRumcajs — APPROVED** [review]

**Review by lemmih — APPROVED** [review]
Maybe look-back instead of `lookback`.

File: .config/forest.dic (hunk 3 above)
Comment (elmattic) [reply .config/forest.dic]: Let keep this in mind and try again in the future. This shouldn't block us.

File: .config/forest.dic [hunk 8]
```diff
@@ -1,21 +1,29 @@
...
+CID
+CIDs
+Filecoin
+IPLD
+JSON
+JWT
+Merkle
+RPC
+SECP
+Secp256k1
```
Comment (lemmih) [inline .config/forest.dic:16]: We should consistently use one or the other.

File: blockchain/chain/src/store/chain_store.rs (hunk 7 above)
Comment (elmattic) [reply blockchain/chain/src/store/chain_store.rs:667]: Hmm.. I will keep `db` for now. I feel `db` is more an abbreviation for the word database. Plus it's also used [here](https://github.com/ChainSafe/forest/blame/main/scripts/sync_check/docker-compose.yml#L21) 😁

File: .config/forest.dic (hunk 8 above)
Comment (elmattic) [reply .config/forest.dic:16]: So which one do we keep? I feel `SECP` is best in our context.

Comment (LesnyRumcajs) [reply .config/forest.dic:16]: In general, both are used in Filecoin specs, also in a variant of `secpk`.  
https://spec.filecoin.io/

File: .config/spellcheck.toml [hunk 9]
```diff
@@ -10,7 +10,12 @@ tokenization_splitchars = "\",;:.!?#(){}[]|/_-‒'`&@§¶…"
...
+  # single char `=` `>` `%` ..
+  "^=|>|<|%$",
+]
```
Comment (lemmih) [inline .config/spellcheck.toml:18]: I'm not sure we should allow these characters. If we have equations in the documentation, they should be marked as code snippets.

File: blockchain/chain/src/store/chain_store.rs [hunk 10]
```diff
@@ -778,8 +778,8 @@ where
         .transpose()?)
 }
 
-/// Attempts to deserialize to unsigend message or signed message and then returns it at as a
-// [ChainMessage].
+/// Attempts to de-serialize to unsigned message or signed message and then returns it as a
```
Comment (lemmih) [inline blockchain/chain/src/store/chain_store.rs:781]: I'm so happy when typos like this are fixed. It's like we're real software now. :)

Comment (elmattic) [reply blockchain/chain/src/store/chain_store.rs:781]: Exactly, and good job for the setup of this tool!
